sszgen/generator/unmarshal.go

```diff
@@ -340,7 +340,7 @@ func (v *Value) createSlice(useNumVariable bool) string {
 		// []int uses the Extend functions in the fastssz package
 		return fmt.Sprintf("::.%s = ssz.Extend%s(::.%s, %s)", v.name, uintVToName(v.e), v.name, size)
 
-	case TypeContainer:
+	case TypeContainer, TypeReference:
```

Comment:
I'd add some tests to current test suite (it will be checked inside our project but anyway...)

Also seems we need to fix error messages for this case...

https://github.com/NilFoundation/fastssz/blob/93b9cf0cc18b17022187be8551735957f3b82eca/encode.go#L303-L309

Reply:
What is the error message?

Reply:
Seems in case if user passes malformed input or adds more elements than limit allows it will get "xx" or "yy" as an error message. 

It's not related to your patch directly, it's just interesting moment.

Reply:
I see, let's do it later.
About tests, I tried to find a place in fastssz where to make them, but no luck. Let's push it as is and see what we'll need.